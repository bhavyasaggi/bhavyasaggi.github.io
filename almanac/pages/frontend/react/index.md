---
title: React
eleventyNavigation:
  parent: /pages/frontend/
  order: 21
---

# React

## Styling Libraries

### Stylesheet Generators

- Less / Sass / Scss
- bootstrap
- tailwind
- vanilla-extract

### Styled Components

- css-modules
- styled-components
- emotion.js

## Iconography

## Component Libraries

- React-Bootstrap
- Ant Design
- MUI

## Functionality Libraries

- Formik
- Framer Motion
- RxJS
