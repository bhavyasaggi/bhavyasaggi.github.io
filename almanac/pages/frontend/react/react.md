---
title: React
eleventyNavigation:
  parent: /pages/frontend/
  order: 21
---

# React

## Styling Libraries

### Preprocessors (Less / Sass / Scss)

### Stylesheet Generators

- bootstrap
- tailwind
- vanilla-extract

### Styled Components

- css-modules
- styled-components
- emotion.js

## Iconography

## Component Libraries

- React-Bootstrap
- Ant Design
- MUI

## Functionality Libraries

- Formik
- Framer Motion
- RxJS
