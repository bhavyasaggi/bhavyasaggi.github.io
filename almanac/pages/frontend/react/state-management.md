---
title: State Management
eleventyNavigation:
  parent: /pages/frontend/react/
  order: 1
---

# State Management

## Global Store

- Redux

## Atomic State

- Recoil / jotai

- Zustand

## Query Based

- RTK / Apollo
