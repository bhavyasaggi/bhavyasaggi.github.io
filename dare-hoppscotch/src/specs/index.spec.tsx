import { render } from '@testing-library/react'

import PageHome from '../pages/index'

describe('PageHome', () => {
  it('Snapshot', () => {
    const { container } = render(<PageHome />)
    expect(container).toMatchSnapshot()
  })
})
