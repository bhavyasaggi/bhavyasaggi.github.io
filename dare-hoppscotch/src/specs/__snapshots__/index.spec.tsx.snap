// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHome Snapshot 1`] = `
<div>
  <main
    class="min-vh-100 d-flex align-items-center justify-content-center className"
  >
    <div
      class="text-center"
    >
      <div
        class="display-1"
      >
        Welcome!
      </div>
      <a
        href="/tracker"
      >
        <button
          class="btn btn-dark btn-lg"
          type="button"
        >
          Visit Dashboard
        </button>
      </a>
    </div>
  </main>
</div>
`;
